### ISSUE TITLE
interrupting the kernel while polling (inactive) kills it

### ISSUE THREAD

**Issue by Anne623** [IRkernel/IRkernel#245]
Hi ,All

   when I click interrupt button in notebook page, the r kernel will be dead when there are no cell is running..it will show dialog:

```
Kernel Restarting
The kernel appears to have died. It will restart automatically.
```

but if interrupt a running cell,it works well.

**Comment by flying-sheep** [comment]
is this the git version or a release?

**Comment by takluyver** [comment]
I could replicate it - #246 fixes it for me.

**Comment by flying-sheep** [comment]
ah, it occurs in both pbdZMQ and rzmq versions.

**Comment by Anne623** [comment]
Hi, it not working, i use release V0.5, and i add try catch in 
```
 while (TRUE) {
            tryCatch({
                events <- poll.socket(
                    list(sockets$hb, sockets$shell, sockets$control),
                    list('read', 'read', 'read'), timeout = -1L)
                if (events[[1]]$read) {
                    # heartbeat
                    hb_reply()
                }
                if (events[[2]]$read) {
                    # Shell socket
                    handle_shell()
                }
                if (events[[3]]$read) {  # Control socket
                    handle_control()
                }
            }, interrupt=function(cond) {}
            )
        }
```

it also will be kill kernel:
```
Error in poll.socket(list(sockets$hb, sockets$shell, sockets$control),  : 
  Interrupted system call
Calls: <Anonymous> ... tryCatchList -> tryCatchOne -> doTryCatch -> poll.socket -> .Call
Execution halted
```

**Comment by Anne623** [comment]
@flying-sheep yes,it's release V0.5

**Comment by flying-sheep** [comment]
yes, we’re currently discussing this in #246

**Comment by flying-sheep** [comment]
Fixed in 771c6fb5309d6bd0fc162a73058ebc59dda46389
